include/os/linux/spl/sys/mutex.h

```diff
@@ -128,7 +128,6 @@ spl_mutex_lockdep_on_maybe(kmutex_t *mp)			\
 
 #define	NESTED_SINGLE 1
 
-#ifdef CONFIG_DEBUG_LOCK_ALLOC
```

Comment:
I removed the definition of mutex_enter_nested() with CONFIG_DEBUG_LOCK_ALLOC undefined because I believe it is redundant - when CONFIG_DEBUG_LOCK_ALLOC is not defined, the Linux kernel header mutex.h will redefine mutex_lock_nested() to mutex_lock() anyways. If ZFS is compiled with CONFIG_DEBUG_LOCK_ALLOC  I believe this implies Linux must be also, so this single definition should behave the same in all cases as the previous pair of definitions. Let me know if I'm missing anything, but I think this simplifies the code and reduces duplication.

Reply:
This looks like it should be fine.  I thought the older 3.10 kernel might not define `mutex_lock_nested()` for both CONFIG_DEBUG_LOCK_ALLOC and !CONFIG_DEBUG_LOCK_ALLOC, but it does.